tracer/test/Datadog.Trace.Tests/Configuration/TracerSettingsTests.cs

```diff
@@ -194,6 +198,644 @@ public void SetServerHttpCodes()
+
+            var source = CreateConfigurationSource((ConfigurationKeys.FeatureFlags.OpenTelemetryEnabled, value), (fallbackKey, fallbackValue));
+            var settings = new TracerSettings(source);
+
+            settings.IsActivityListenerEnabled.Should().Be(expected);
+        }
+
+        [Theory]
+        [InlineData("test1,, ,test2", "test3,, ,test4", "test5,, ,test6", new[] { "test1", "test2" })]
+        [InlineData("", "test3,, ,test4", "test5,, ,test6", new[] { "tracecontext", "Datadog" })]
+        [InlineData(null, "test3,, ,test4", "test5,, ,test6", new[] { "test3", "test4" })]
+        [InlineData(null, null, "test5,, ,test6", new[] { "test5", "test6" })]
+        [InlineData(null, null, null, new[] { "tracecontext", "Datadog" })]
+        public void PropagationStyleInject(string value, string legacyValue, string fallbackValue, string[] expected)
+        {
+            const string legacyKey = "DD_PROPAGATION_STYLE_INJECT";
+
+            foreach (var isActivityListenerEnabled in new[] { true, false })
+            {
+                var source = CreateConfigurationSource(
+                    (ConfigurationKeys.PropagationStyleInject, value),
+                    (legacyKey, legacyValue),
+                    (ConfigurationKeys.PropagationStyle, fallbackValue),
+                    (ConfigurationKeys.FeatureFlags.OpenTelemetryEnabled, isActivityListenerEnabled ? "1" : "0"));
+
+                var settings = new TracerSettings(source);
+
+                settings.PropagationStyleInject.Should().BeEquivalentTo(isActivityListenerEnabled ? expected.Concat("tracecontext") : expected);
+            }
+        }
+
+        [Theory]
+        [InlineData("test1,, ,test2", "test3,, ,test4", "test5,, ,test6", new[] { "test1", "test2" })]
+        [InlineData("", "test3,, ,test4", "test5,, ,test6", new[] { "tracecontext", "Datadog" })]
+        [InlineData(null, "test3,, ,test4", "test5,, ,test6", new[] { "test3", "test4" })]
+        [InlineData(null, null, "test5,, ,test6", new[] { "test5", "test6" })]
+        [InlineData(null, null, null, new[] { "tracecontext", "Datadog" })]
+        public void PropagationStyleExtract(string value, string legacyValue, string fallbackValue, string[] expected)
+        {
+            const string legacyKey = "DD_PROPAGATION_STYLE_EXTRACT";
+
+            foreach (var isActivityListenerEnabled in new[] { true, false })
+            {
+                var source = CreateConfigurationSource(
+                    (ConfigurationKeys.PropagationStyleExtract, value),
+                    (legacyKey, legacyValue),
+                    (ConfigurationKeys.PropagationStyle, fallbackValue),
+                    (ConfigurationKeys.FeatureFlags.OpenTelemetryEnabled, isActivityListenerEnabled ? "1" : "0"));
+
+                var settings = new TracerSettings(source);
+
+                settings.PropagationStyleExtract.Should().BeEquivalentTo(isActivityListenerEnabled ? expected.Concat("tracecontext") : expected);
```

Comment:
Looks like this highlights a bug, no?  If no propagation values are set, but `OpenTelemetryEnabled=1`, then the expected is `new[] { "tracecontext", "Datadog", "tracecontext")`? 🤔

Reply:
It's not a bug, there's a comment in TracerSettings saying it's expected and "we'll handle that later"😛  https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Configuration/TracerSettings.cs#L226